packages/admin/admin-stories/src/admin/table/TableInStackWithStackFormAndEditDialogForm.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+    Field,
+    FinalForm,
+    FinalFormInput,
+    SaveButton,
+    Stack,
+    StackBackButton,
+    StackLink,
+    StackPage,
+    StackSwitch,
+    Toolbar,
+    ToolbarActions,
+    ToolbarFillSpace,
+    ToolbarItem,
+    useEditDialog,
+    useStackApi,
+    useStackSwitchApi,
+} from "@comet/admin";
+import { Add as AddIcon, Edit } from "@comet/admin-icons";
+import { Button, Link, Typography } from "@mui/material";
+import { DataGrid } from "@mui/x-data-grid";
+import { storiesOf } from "@storybook/react";
+import * as React from "react";
+import { v4 } from "uuid";
+
+import { apolloRestStoryDecorator } from "../../apollo-rest-story.decorator";
+import { storyRouterDecorator } from "../../story-router.decorator";
+
+interface ExampleRow {
+    id: string;
+    foo: string;
+    bar: string;
+}
+type NewExampleInput = Omit<ExampleRow, "id">;
+
+interface EditFormProps {
+    row?: ExampleRow;
+    mode: "edit" | "add";
+    onSubmit: (newData: ExampleRow | NewExampleInput) => Promise<ExampleRow>;
+}
+function EditForm(props: EditFormProps) {
+    const stackApi = useStackApi();
+    const stackSwitchApi = useStackSwitchApi();
+
+    const newId = React.useRef<string>();
+
+    return (
+        <FinalForm
+            mode={props.mode}
+            initialValues={props.row}
+            onSubmit={async (values) => {
+                const newRow = await props.onSubmit(values);
+                if (props.mode === "add") {
+                    newId.current = newRow.id;
+                }
+            }}
+            onAfterSubmit={() => {
+                if (props.mode === "edit") {
+                    stackApi?.goBack();
+                } else {
+                    if (newId.current) {
+                        stackSwitchApi.activatePage("edit", newId.current);
+                    }
```

Comment:
> Shouldn't this navigation happen in the new `onAfterSubmit` handler? 

It could also happen there, but usually the form will be in a child component (like in my case) and passing the id up is tedious.

> Or could there be other use cases for the callback?

I'm not sure but I think there might. Since it's just one function call, I would keep it (no maintenance work)